internal/component/mimir/rules/kubernetes/rules.go

```diff
@@ -239,6 +294,26 @@ func (c *Component) Update(newConfig component.Arguments) error {
 	return <-errChan
 }
 
+func (c *Component) NotifyClusterChange() {
+	c.clusterUpdates <- struct{}{}
+}
```

Comment:
When we talked about this offline, we discussed that there will eventually be a transition to per-component scheduling where the user would opt-in components to this type of behavior. Moving to per-component scheduling could cause bugs here; notably we would want to ignore these updates if per-component scheduling was configured, since the ownership calculation of per-component scheduling may not work the same as the ownership detection here, and if both are in play there may be no Alloy instances running the component. 

Can we add a comment here notifying us of this future issue?